doc/manual_experimental.rst

```diff
@@ -1990,6 +1990,24 @@ A borrow starts with one of the following:
 - The assignment of a location that is derived from a local parameter
   to a view-type.
 
+If a non l-value is borrowed from, as in:
+
+.. code-block:: nim
+
+  func f(): Obj
+  let v: lent Obj = f()
```

Comment:
this looks pretty dangerous - the whole idea behind explicitly typing out `lent` is that a copy is expensive or wrong and you want certain semantics to be guaranteed

Reply:
But there is no copy here, we borrow from a temporary location and `f()`'s result is moved/bitcopied into `tmp` or even constructed directly inside `tmp` due to NRVO.

Reply:
However, you're right that maybe it's not "non l-value" but instead "routine call".

Reply:
> moved/bitcopied

this is the risk: for large objects, there will be a significant cost and stack space usage involved - also, nrvo works only sometimes

Reply:
Well we should really fix NRVO. But I don't see the connection, when you call `f()` and it produces a huge value, that call doesn't suddenly get worse performance-wise just because you borrow from its value.

Reply:
well, if `f()` gives me something unborrowable, I'd want it to stop me, not introduce a magic temporary on the stack

Reply:
well, I agree but @zah does not. ;-)